src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -1383,6 +1384,169 @@ srcDevices=#newSrcEnabled, dstDevices=#newDstEnabled"() {
         1                 | 3                | 5                | FlowEncapsulationType.VXLAN
     }
 
+    @Tidy
+    def "Switch is not containing extra rules after connected devices removal"() {
+
+        given: "A switch with devices feature turned on"
+        assumeTrue(topology.activeTraffGens.size() > 0, "Require at least 1 switch with connected traffgen")
+        def tg = topology.activeTraffGens[0]
+        def sw = tg.switchConnected
+        def initialProps = switchHelper.getCachedSwProps(sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialProps.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = true
+            it.switchArp = true
+        })
+
+        and: "Flow is created on a target switch with devices feature 'off'"
+        def dst = topology.activeSwitches.find { it.dpId != sw.dpId }
+        def flow = flowHelperV2.randomFlow(sw, dst)
+        flow.source.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flow.destination.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flowHelperV2.addFlow(flow)
+
+        when: "Turn LLDP and ARP detection off on switch"
+        switchHelper.updateSwitchProperties(sw, initialProps.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = false
+            it.switchArp = false
+        })
+        flow.maximumBandwidth = flow.maximumBandwidth + 1
+        flowHelperV2.updateFlow(flow.flowId, flow)
+
+        then: "Check excess rules are not registered on device"
+        wait(WAIT_OFFSET) {
+             verifySwitchRules(sw.dpId)
+        }
+        cleanup: "Remove created flow and registered devices, revert switch props"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        sw && database.removeConnectedDevices(sw.dpId)
```

Comment:
same applies for other new tests, especially if they have 'assumeTrue' before cleanup variable is initialized